docs/basic-features/data-fetching/data-fetching.md

```diff
@@ -0,0 +1,165 @@
+
+The following example shows how to use the `revalidate` property:
+
+```jsx
+function Blog({ posts }) {
+  return (
+    <ul>
+      {posts.map((post) => (
+        <li>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+
+// This function gets called at build time on server-side.
+// It may be called again, on a serverless function, if
+// revalidation is enabled and a new request comes in
+export async function getStaticProps() {
+  const res = await fetch('https://.../posts')
+  const posts = await res.json()
+
+  return {
+    props: {
+      posts,
+    },
+    // Next.js will attempt to re-generate the page:
+    // - When a request comes in
+    // - At most once every 10 seconds
+    revalidate: 10, // In seconds
+  }
+}
+
+// This function gets called at build time on server-side.
+// It may be called again, on a serverless function, if
+// the path has not been generated.
+export async function getStaticPaths() {
+  const res = await fetch('https://.../posts')
+  const posts = await res.json()
+
+  // Get the paths we want to pre-render based on posts
+  const paths = posts.map((post) => ({
+    params: { id: post.id },
+  }))
+
+  // We'll pre-render only these paths at build time.
+  // { fallback: blocking } will server-render pages
+  // on-demand if the path doesn't exist.
+  return { paths, fallback: 'blocking' }
+}
+
+export default Blog
+```
+
+When a request is made to a page that was pre-rendered at build time, it will initially show the cached page.
+
+- Any requests to the page after the initial request and before 10 seconds are also cached and instantaneous.
+- After the 10-second window, the next request will still show the cached (stale) page
+- Next.js triggers a regeneration of the page in the background.
+- Once the page has been successfully generated, Next.js will invalidate the cache and show the updated product page. If the background regeneration fails, the old page remains unaltered.
+
+When a request is made to a path that hasn’t been generated, Next.js will server-render the page on the first request. Future requests will serve the static file from the cache.
+
+[Incremental Static Regeneration](https://vercel.com/docs/next.js/incremental-static-regeneration) covers how to persist the cache globally and handle rollbacks.
+
+## SWR
+
+The team behind Next.js has created a React hook for data fetching called [**SWR**](https://swr.vercel.app/). It is highly recommend if you are fetching data on the client side. It handles caching, revalidation, focus tracking, refetching on interval, and more.
+
+```jsx
+import useSWR from 'swr'
+
+function Profile() {
+  const { data, error } = useSWR('/api/user', fetch)
+
+  if (error) return <div>failed to load</div>
+  if (!data) return <div>loading...</div>
+  return <div>hello {data.name}!</div>
+}
+```
+
+[Check out the SWR documentation to learn more](https://swr.vercel.app/).
+
+## Learn more
+
+We recommend you to read the following sections next:
+
+<div class="card">
```

Comment:
Let's make these cards the main focus on the page maybe? 

- CSR
- SSR
- ISR
- SSG